### PR TITLE
WIP: Optimize package management [skip ci]

### PR DESCRIPTION
**Pull request by nordlow**

### CODE REVIEW

**Comment by nordlow** [comment]
CI seems broken.

**Comment by nordlow** [comment]
Ready for review now, @Geod24.

**Comment by nordlow** [comment]
Changed to WIP. I just realized yet another optimization.

File: source/dub/packagemanager.d [hunk 1]
```diff
@@ -754,27 +754,32 @@ class PackageManager {
...
+			logDebug("iterating dir %s", path.toNativeString());
+			try
+				foreach (DirEntry dent; dirEntries(path.toNativeString(), SpanMode.shallow)) {
+					logDebug("iterating dir %s entry %s", path.toNativeString(), dent.name);
+					if (!dent.isDir) continue;
 
 					// Old / flat directory structure, used in non-standard path
-					// Packages are stored in $ROOT/$SOMETHING/`
-					auto pack_path = path ~ (pdir.name ~ "/");
+					// Packages are stored in $ROOT/$NAME-$VERSION/`
```
Comment (nordlow) [inline source/dub/packagemanager.d:765]: Do we still need to support file structure, @Geod24?

**Comment by Geod24** [comment]
This I need to review from a computer, but reducing the number of scans shouldn't be the goal. Eliminating the scan all together is IMO the way to go.

**Comment by nordlow** [comment]
Are CI failure unrelated?

**Comment by Geod24** [comment]
> Are CI failures unrelated?

Why would they be unrelated ?

**Review by WebFreak001 — CHANGES_REQUESTED** [review]
this includes non-trivial changes to source code, please rebase and remove the `[skip ci]`

**Comment by Geod24** [comment]
We now have a no-scan behavior, closing as stalled.

**Comment by nordlow** [comment]
> We now have a no-scan behavior, closing as stalled.

Awesome. Thanks.

**Pull request closed without merging**